### ISSUE TITLE
Index buffers don't render on Safari WebGL

### ISSUE THREAD

**Issue by willmcpherson2** [gfx-rs/wgpu#1535]
Apologies for the not-so-minimal example, I was following along with Learn Wgpu and noticed my code stopped working with [index buffers](https://sotrh.github.io/learn-wgpu/beginner/tutorial4-buffer/#the-index-buffer).

Also I understand that dealing with Safari's quirks is not so fun, but I was hoping I could at least get some insight on what to do here.

Sorry if my code is wrong!

**Description**

The WebGL backend on Safari doesn't render vertices when using an index buffer. The same code runs perfectly fine on Firefox.

**Repro steps**

Download this complete example:

https://github.com/willmcpherson2/wu

Build and serve:

```sh
cargo build --target wasm32-unknown-unknown
wasm-bindgen --out-dir static --target web --no-typescript target/wasm32-unknown-unknown/debug/wu.wasm
python3 -m http.server -d static
```

You can then remove the index buffer, which fixes the issue:

```sh
git checkout 5415d390bc65b26338de1108082828fb34b83718
```

**Expected vs observed behavior**

Expected a triangle to render. Correctly renders on Firefox 89.0.1 and Metal. Does not render on Safari 14.1.1.

**Extra materials**

[Firefox log](https://gist.github.com/willmcpherson2/c52f8d98b609e0d1efcfe505f1161d0b#file-firefox-log)

[Firefox screenshot](https://i.imgur.com/df82iY1.png)

[Safari log](https://gist.github.com/willmcpherson2/c52f8d98b609e0d1efcfe505f1161d0b#file-safari-log)

[Safari screenshot](https://i.imgur.com/HonaBnN.png)

**Platform**

- Apple M1 Macbook Air
- macOS Big Sur 11.4
- Safari 14.1.1 (WebGL 2.0 enabled)
- [Cargo.toml](https://github.com/willmcpherson2/wu/blob/main/Cargo.toml)

**Comment by kvark** [comment]
According to https://caniuse.com/webgl2, you can use WebGL2 starting Safari 15.
Our WebGL backend is based on GLES3, so we require WebGL2.

**Comment by kvark** [comment]
Interesting... I don't know what's up there. Could be bugs in Safari, given that WebGL2 for them is fresh new?

**Comment by FrankenApps** [comment]
This is fixed for me in Safari version `15.2`.
Possibly earlier (e.g. `15.1`), but I did not check any further.
